dask_expr/shuffle.py

```diff
@@ -127,7 +132,7 @@ def from_abstract_shuffle(cls, expr: Shuffle) -> Expr:
         partitioning_index = expr.partitioning_index
         npartitions_out = expr.npartitions_out
         ignore_index = expr.ignore_index
-        options = expr.options
+        options = expr.options or {}
```

Comment:
Nit: maybe normalize this ahead of time rather than here?  Maybe even in `_defaults` (although I know using a mutable default is sometimes worrisome).